### DISCUSSION TITLE
CDP listener is not working in multi-thread environment

### DISCUSSION THREAD

**Discussion by MovingW** [seleniumbase/SeleniumBase#2214] [Q&A] [answered]
I found one issue. The cdp listener is not working in multi-thread environment. I put the webdriver instance in message_lisnener() and run it with threading, but it turned out that the callback func of the listener is not called at all.

`t1 = threading.Thread(target=message_listener)
 t1.start()
`

I put a print() function here but there's nothing printed.

`def __add_cdp_listeners(self, webdriver):     
     webdriver.add_cdp_listener(
         "Network.requestWillBeSent",
         lambda data: print(data)
     )
`

**Comment by mdmintz** [comment, marked as answer]
Upgrade to `4.20.8`. See the new options: https://github.com/seleniumbase/SeleniumBase/issues/2220
